test/integ/test_session_parameters.py

```diff
@@ -25,3 +25,72 @@ def test_session_parameters(db_parameters):
     ret = connection.cursor().execute(
         "show parameters like 'TIMEZONE'").fetchone()
     assert ret[1] == 'UTC'
+
+
+def test_client_session_keep_alive(db_parameters):
+    """Tests client_session_keep_alive setting.
+
+    Ensures that client's explicit config for client_session_keep_alive
+    session parameter is always honored and given higher precedence over
+    user and account level backend configuration.
+    """
+    # Ensure backend parameter is set to False
+    set_backend_client_session_keep_alive(db_parameters, False)
+    connection = create_client_connection(db_parameters, True)
+
+    ret = connection.cursor().execute(
+        "show parameters like 'CLIENT_SESSION_KEEP_ALIVE'").fetchone()
+    assert ret[1] == 'true'
+    connection.close()
+
+    # Set backend parameter to True
+    set_backend_client_session_keep_alive(db_parameters, True)
+
+    # Set session parameter to False
+    connection = create_client_connection(db_parameters, False)
+    ret = connection.cursor().execute(
+        "show parameters like 'CLIENT_SESSION_KEEP_ALIVE'").fetchone()
+    assert ret[1] == 'false'
+    connection.close()
+
+    # Set session parameter to None backend parameter continues to be True
+    connection = create_client_connection(db_parameters, None)
+    ret = connection.cursor().execute(
+        "show parameters like 'CLIENT_SESSION_KEEP_ALIVE'").fetchone()
+    assert ret[1] == 'true'
+    connection.close()
+
+
+def create_client_connection(db_parameters: object, val: bool) -> object:
+    """Create connection with client session keep alive set to specific value."""
+    connection = snowflake.connector.connect(
+        protocol=db_parameters['protocol'],
+        account=db_parameters['account'],
+        user=db_parameters['user'],
+        password=db_parameters['password'],
+        host=db_parameters['host'],
+        port=db_parameters['port'],
+        database=db_parameters['database'],
+        schema=db_parameters['schema'],
+        client_session_keep_alive=val
+    )
+    return connection
+
+
+def set_backend_client_session_keep_alive(db_parameters: object, val: bool) -> None:
```

Comment:
I'd change that this function takes a connection from the outside, the less round trips we need to do to Snowflake the better